src/Sulu/Bundle/ContentBundle/Controller/ContentController.php

```diff
@@ -0,0 +1,347 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ContentBundle\Controller;
+
+use FOS\RestBundle\Routing\ClassResourceInterface;
+use FOS\RestBundle\View\ViewHandlerInterface;
+use Hateoas\Representation\CollectionRepresentation;
+use PHPCR\ItemNotFoundException;
+use Sulu\Component\Content\Repository\Content;
+use Sulu\Component\Content\Repository\ContentRepositoryInterface;
+use Sulu\Component\Content\Repository\Mapping\MappingBuilder;
+use Sulu\Component\Content\Repository\Mapping\MappingInterface;
+use Sulu\Component\PHPCR\SessionManager\SessionManagerInterface;
+use Sulu\Component\Rest\Exception\MissingParameterException;
+use Sulu\Component\Rest\Exception\ParameterDataTypeException;
+use Sulu\Component\Rest\RequestParametersTrait;
+use Sulu\Component\Rest\RestController;
+use Sulu\Component\Security\Authentication\UserInterface;
+use Sulu\Component\Webspace\Manager\WebspaceManagerInterface;
+use Sulu\Component\Webspace\Webspace;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * Provides api for content querying.
+ */
+class ContentController extends RestController implements ClassResourceInterface
+{
+    const WEBSPACE_NODE_SINGLE = 'single';
+    const WEBSPACE_NODES_ALL = 'all';
+
+    private static $relationName = 'content';
+
+    use RequestParametersTrait;
```

Comment:
I think we don't have a guide line for this until now, but I would put the used traits at the very beginning of the class, since it is more similiar to the extends and implements clause. Also it is very important for me as a developer to know, because it adds more functions to this class. /cc @chirimoya